core/src/mast/merger/mod.rs

```diff
@@ -163,6 +167,17 @@ impl MastForestMerger {
         Ok(())
     }
 
+    fn merge_advice_map(&mut self, other_forest: &MastForest) -> Result<(), MastForestError> {
+        for (key, value) in other_forest.advice_map.clone().into_iter() {
+            if self.mast_forest.advice_map().get(&key).is_some() {
```

Comment:
It would be a bit more forgiving to only return an error if the advice maps have different values at the same key

Reply:
Good point! Done. 
My line of thinking was that keys in the advice map are expected to be so unique (hashes) that any collision would mean that something is off and we better fail early. I suppose we can cut some slack when values are the same (user sets the values twice?).